test_runner/src/main/kotlin/ftl/run/status/ExecutionStatusPrinter.kt

```diff
@@ -0,0 +1,86 @@
+package ftl.run.status
+
+import com.google.common.annotations.VisibleForTesting
+import ftl.args.IArgs
+import ftl.args.outputStyle
+import ftl.config.FtlConstants
+import org.fusesource.jansi.Ansi
+import org.fusesource.jansi.AnsiConsole
+
+fun createExecutionStatusPrinter(
+    args: IArgs
+): (List<ExecutionStatus.Change>) -> Unit = when (args.outputStyle()) {
+    OutputStyle.Multi -> MultiLinePrinter()
+    OutputStyle.Single -> SingleLinePrinter()
+    OutputStyle.Verbose -> VerbosePrinter
+}
+
+@VisibleForTesting
+internal class SingleLinePrinter : (List<ExecutionStatus.Change>) -> Unit {
+    private var previousLineSize = 0
+    override fun invoke(changes: List<ExecutionStatus.Change>) {
+        val time = changes.firstOrNull()?.time ?: return
+        changes.fold(emptyMap<String, Int>()) { acc, (_, _, current: ExecutionStatus) ->
+            acc + Pair(
+                first = current.state,
+                second = acc.getOrDefault(current.state, 0) + 1
+            )
+        }.toList().joinToString(", ") { (status, count) ->
+            "$status:$count"
+        }.let { statusCounts ->
+            print("\r" + (0..previousLineSize).joinToString("") { " " })
+            val output = "${FtlConstants.indent}$time Test executions status: $statusCounts"
+            previousLineSize = output.length
+            print("\r$output")
+        }
+    }
+}
+
+@VisibleForTesting
+internal class MultiLinePrinter(
+    private val ansi: () -> Ansi = { Ansi.ansi() }
+) : (List<ExecutionStatus.Change>) -> Unit {
+    init {
+        AnsiConsole.systemInstall()
+    }
+
+    private val output = LinkedHashMap<String, ExecutionStatus.View>()
+    override fun invoke(changes: List<ExecutionStatus.Change>) {
+        repeat(output.size) {
+            print(ansi().cursorUpLine().eraseLine().toString())
+        }
+        output += changes.map { change ->
+            listExecutionStatusView(change).takeLast(1)
+        }.flatten().map { view ->
+            view.id to view
+        }
+        output.values.joinToString("\n").let(::println)
+    }
+}
+
+@VisibleForTesting
+internal object VerbosePrinter : (List<ExecutionStatus.Change>) -> Unit {
+    override fun invoke(changes: List<ExecutionStatus.Change>) {
+        changes.map(listExecutionStatusView).flatten().forEach(::println)
+    }
+}
+
+private val listExecutionStatusView: (ExecutionStatus.Change) -> List<ExecutionStatus.View>
```

Comment:
How about extension property here instead? Something like (just proposal, not change request :) ):
```
private val ExecutionStatus.Change.views: List<ExecutionStatus.View>
    get() = current.progress
        .minus(previous.progress)
        .map { ExecutionStatus.View(time, name, it) }
        .let { list ->
            if (current.error != previous.error && current.error != null)
                list + ExecutionStatus.View(time, name, "Error: ${current.error}")
            else list
        }
        .let { list ->
            if (current.state != previous.state)
                list + ExecutionStatus.View(time, name, current.state)
            else list
        }
```

Let me know what you think?

Reply:
It doesn't make much difference, but I can change it.

Reply:
IMO it's more clear when I see sth like
`change.views.takeLast`
but I don't want to force anything.